tools/kicad_rs/src/parser.rs

```diff
@@ -1,23 +1,43 @@
 use std::collections::HashMap;
 use std::error::Error;
-use std::fmt;
 use std::path::Path;
 
 use kicad_parse_gen::schematic as kicad_schematic;
 
+use crate::error::errorf;
 use crate::types::*;
 
+type ParseResult<T> = Result<T, Box<dyn Error>>;
+
 impl Schematic {
-    pub fn parse(p: &Path) -> Result<Schematic, Box<dyn Error>> {
+    pub fn parse(p: &Path) -> ParseResult<Schematic> {
         parse_schematic(p, String::new())
     }
 }
 
-// parse_schematic turns a schematic at path p into the recursive Schematic struct.
-fn parse_schematic(p: &Path, id: String) -> Result<Schematic, Box<dyn Error>> {
+/// Turns a KiCad schematic at `path` into a recursive Schematic struct
+fn parse_schematic(path: &Path, id: String) -> ParseResult<Schematic> {
     // Read the schematic using kicad_parse_gen
-    let kisch = kicad_parse_gen::read_schematic(p)?;
+    let kisch = kicad_parse_gen::read_schematic(path)?;
+
+    // Parse the fields for the schematic
+    let meta = parse_meta(&kisch, path)?;
```

Comment:
None if these 4 seem to actually require the "whole" of a `kicad_parse_gen::Schematic`. Maybe you can scope this down to just the description, components vector, etc.?

Reply:
Ok, I have no strong preference. It's good as-is